pymeasure/instruments/keithley/keithley6487.py

```diff
@@ -0,0 +1,395 @@
+        and measurement speed. Takes values from 0.001 to 6.0 or 5.0, where 6.0
+        is 60Hz and 5.0 is 50Hz"""
+    )
+
+    ###########
+    # Trigger #
+    ###########
+
+    trigger_count = Instrument.control(
+        ":TRIG:COUN?", ":TRIG:COUN %d",
+        """ An integer property that controls the trigger count,
+        which can take values from 1 to 2048. """,
+        validator=truncated_range,
+        values=[1, 2048],
+        cast=int
+    )
+    trigger_count_inf = Instrument.control(
+        ":TRIG:COUNT?", "TRIG:COUNT %s",
+        """A string that controls the trigger count,
+        sets trigger count to INF mode instead of an integer""",
+        validator=strict_discrete_set,
+        values=['INF'],
+        cast=str
+    )
+
+    trigger_delay = Instrument.control(
+        ":TRIG:SEQ:DEL?", ":TRIG:SEQ:DEL %f",
+        """ A floating point property that controls the trigger delay
+        in seconds, which can take values from 0 to 999.9999 s. """,
+        validator=truncated_range,
+        values=[0, 999.9999]
+    )
+
+    ###########
+    # Filters #
+    ###########
+
+    filter_type = Instrument.control(
+        ":SENS:AVER:TCON?", ":SENS:AVER:TCON %s",
+        """ A String property that controls the filter's type.
+        REP : Repeating filter
+        MOV : Moving filter""",
+        validator=strict_discrete_set,
+        values=['REP', 'MOV'],
+        map_values=False)
+
+    filter_count = Instrument.control(
+        ":SENS:AVER:COUNT?", ":SENS:AVER:COUNT %d",
+        """ A integer property that controls the number of readings that are
+        acquired and stored in the filter buffer for the averaging""",
+        validator=truncated_range,
+        values=[2, 100],
+        cast=int)
+
+    filter_state = Instrument.control(
+        ":SENS:AVER?", ":SENS:AVER %s",
+        """ A string property that controls if the filter is active.""",
+        validator=strict_discrete_set,
+        values=['ON', 'OFF'],
+        map_values=False)
+
+    ####################
+    #   Methods        #
+    ####################
+
+    def __init__(self, adapter, **kwargs):
+        super().__init__(
+            adapter, "Keithley 6487 Picoammeter", **kwargs
+        )
+
+    def enable_source(self):
+        """Enables the source voltage of the instrument"""
+        self.write(":SOUR:VOLT:STAT ON")
+
+    def disable_source(self):
+        """Disables the source voltage of the instrument"""
+        self.write(":SOUR:VOLT:STAT OFF")
+
+    def measure_current(self, nplc=1, current=1.05e-4, auto_range=True):
+        """ Configures the measurement of current.
+
+        :param nplc: Number of power line cycles (NPLC) from 0.01 to 10
+        :param current: Upper limit of current in Amps, from -0.021 A to 0.021 A
+        :param auto_range: Enables auto_range if True, else uses the set current
+        """
+        log.info("%s is measuring current." % self.name)
+        self.write(":SENS:FUNC:CURR;"
```

Comment:
Is it possible to rewrite this sequence of commands using defined properties ?
If possible, I think code will be more elegant.
Similar comment for other methods.